meinberlin/apps/projects/api.py

```diff
@@ -39,43 +41,55 @@ def get_queryset(self):
         )
         return projects
 
+    def list(self, request, *args, **kwargs):
+        return caching.add_or_serialize(
+            namespace=request.path,
+            suffix=self.request.GET.get("status"),
+            view_set=self,
+        )
+
     def get_serializer(self, *args, **kwargs):
         if "status" in self.request.GET:
-            statustype = self.request.GET["status"]
-            if statustype == "activeParticipation":
+            status = self.request.GET["status"]
+            if status == "activeParticipation":
                 return project_serializers.ActiveProjectSerializer(
                     now=self.now, *args, **kwargs
                 )
-            if statustype == "futureParticipation":
+            if status == "futureParticipation":
                 return project_serializers.FutureProjectSerializer(
                     now=self.now, *args, **kwargs
                 )
-            if statustype == "pastParticipation":
+            if status == "pastParticipation":
                 return project_serializers.PastProjectSerializer(
                     now=self.now, *args, **kwargs
                 )
         return project_serializers.ProjectSerializer(now=self.now, *args, **kwargs)
 
 
-class PrivateProjectListViewSet(viewsets.ReadOnlyModelViewSet):
+class PrivateProjectViewSet(viewsets.ReadOnlyModelViewSet):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         now = timezone.now()
         self.now = now
 
     def get_queryset(self):
-        private_projects = Project.objects.filter(
-            is_draft=False, is_archived=False, access=Access.PRIVATE
+        filter_kwargs = dict(is_draft=False, is_archived=False, access=Access.PRIVATE)
+        projects = caching.add_or_query(
+            namespace=reverse("projects-list"),
```

Comment:
this should be privateprojects. Better just give hardcoded values everywhere you call these functions, not dependent on url pattern

Reply:
done